aten/src/THCUNN/generic/MultiLabelMarginCriterion.cu

```diff
@@ -58,49 +58,52 @@ void THNN_(MultiLabelMarginCriterion_updateOutput)(
   }
   else if(input->dim() == 2)
   {
-    int nframe = input->size(0);
-    int dim = input->size(1);
-    dim3 blocks(input->size(0));
-    dim3 threads(MULTILABELMARGIN_THREADS);
-
-    if (reduction != at::Reduction::None)
-    {
-      THCTensor *output_tmp = THCTensor_(newWithSize1d)(state, input->size(0));
-      THCTensor_(resize0d)(state, output);
-
-      cunn_MultiLabelMarginCriterion_updateOutput_kernel<scalar_t, accreal>
-        <<<blocks, threads, 0, c10::cuda::getCurrentCUDAStream()>>>(
-          THCTensor_(data)(state, output_tmp),
-          THCTensor_(data)(state, input),
-          THCIndexTensor_(data)(state, target),
-          THCTensor_(data)(state, istarget),
-          nframe, dim,
-          reduction == at::Reduction::Mean
-          );
-      THCudaCheck(cudaGetLastError());
-      auto t = THTensor_wrap(output_tmp);
-      auto r = THTensor_wrap(output);
-      at::native::sum_out(r, t, at::IntArrayRef(std::vector<int64_t>{}), false, r.scalar_type());
-      THCTensor_(free)(state, output_tmp);
-    }
-    else
-    {
-      THCTensor_(resize1d)(state, output, input->size(0));
-
-      cunn_MultiLabelMarginCriterion_updateOutput_kernel<scalar_t, accreal>
-        <<<blocks, threads, 0, c10::cuda::getCurrentCUDAStream()>>>(
-          THCTensor_(data)(state, output),
-          THCTensor_(data)(state, input),
-          THCIndexTensor_(data)(state, target),
-          THCTensor_(data)(state, istarget),
-          nframe, dim,
-          false
-          );
-      THCudaCheck(cudaGetLastError());
+    if (input->numel() != 0) {
```

Comment:
It seems like this check should be `input->numel() == 0` and go around line 34 to short-circuit quickly and avoid the frees later. That would also minimize the diff size here.

Reply:
Thanks for that. Have changed.